src/main/kotlin/interactor/GetCostlierCityWithRentingAnApartmentBetweenCityCenterAndOutsideCityCenterInteractor.kt

```diff
@@ -0,0 +1,46 @@
+package interactor
+
+import interactor.utils.BedroomOption
+import model.CityEntity
+import kotlin.math.abs
+
+
+
+class GetCostlierCityWithRentingAnApartmentBetweenCityCenterAndOutsideCityCenterInteractor(
+    private val dataSource: CostOfLivingDataSource
+) {
+
+    fun execute(option: BedroomOption): CityEntity {
+        return when (option) {
+            BedroomOption.ONE_BEDROOM -> getOneBedroom()
+            BedroomOption.THREE_BEDROOM -> getThreeBedroom()
+        }
+    }
+
+    private fun getThreeBedroom(): CityEntity {
+
```

Comment:
You can split the part of the validation to separate functions so you make the code more readable 
@AmeerAmjed what do you think?

Reply:
ok